code.js

```diff
@@ -1,9 +1,9 @@
 function binarySearch(list, element) {
     if(list.length<1) return -1;//when list = [], return -1
-    list.sort();
+    list.sort(function(a, b){return a - b});
```

Comment:
You shouldn't sort the list inside the binary search. This will make everything much slower.

Reply:
Ok, I removed it.